Update requirements to have pyscf==1.7.2 and no scipy pinning

antalszava opened the pull request:
**Summary:**
Following up on https://github.com/PennyLaneAI/pennylane/pull/926 in PennyLane-QChem, we would like to use similar requirements for the `pyscf` and the `scipy` packages.

josh146 commented:
Thanks @antalszava! If all demos pass (with the exception of `tutorial_vqe_parallel`, which we know has a bug), then feel free to close this PR, and update the other PR in the PL repo with the result.

antalszava commented:
Added the link to the other PR and reverted incrementing the version numbers for the cache names in the build.

Would we still like to change the requirements of `scipy` and `pyscf` here? Removed the `scipy` pinning and pinned  `pyscf==1.7.3` (would be pulled as latest for `pyscf<1.7.4`) for now (can close though if I missed something).

josh146 commented:
> Would we still like to change the requirements of scipy and pyscf here?

If everything passes, we might as well. I don't see a drawback to incrementing versions here if we know it doesn't cause any issues.

antalszava commented on requirements.txt:
```diff
@@ -10,7 +10,7 @@ strawberryfields==0.15
 pillow==6.2.0
 pip==20.1
-pyscf==1.7.1
+pyscf==1.7.3
```
```suggestion
pyscf==1.7.2
```

antalszava commented:
The latest failed check contained the known problem with `tutorial_vqe_parallel`:
https://app.circleci.com/pipelines/github/PennyLaneAI/qml/1947/workflows/aedf67f3-7718-47a2-b9ef-78b560d57ec1/jobs/2018

Reverting the caching numbers.
